cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -310,10 +315,22 @@ func generateMountsModeRecover(mst *initramfsMountsState) error {
 	if err := modeEnv.WriteTo(boot.InitramfsWritableDir); err != nil {
 		return err
 	}
-	// we need to put the file to disable cloud-init in the
-	// _writable_defaults dir for writable-paths(5) to install it properly
-	writableDefaultsDir := sysconfig.WritableDefaultsDir(boot.InitramfsWritableDir)
-	if err := sysconfig.DisableCloudInit(writableDefaultsDir); err != nil {
+
+	_, essSnaps, err := mst.ReadEssential("", []snap.Type{snap.TypeGadget})
```

Comment:
this will re-validate all assertions, we need to get the gadgetSnap out of the ReadEssential in the common code I think